### PR TITLE
Add integration test for multiple storage targets

### PR DESCRIPTION
**Pull request by tohhsinpei**
### Description
This PR adds an integration test for #4281, verifying that #3390 is indeed fixed. The test does the following:
- Apply the targets `main` and `other` to two different Storage buckets
- Run the Storage emulator with different rules files configured for the targets (`storage.main.rules` and `storage.other.rules` respectively)
- Check that different rules are applied for the same request to different buckets

### Sample Commands
`npm run test:storage-emulator-integration` will run the usual integration test `scripts/storage-emulator-integration/tests.ts` as well as this new integration test.

### CODE REVIEW

File: scripts/storage-emulator-integration/multiple-targets/storage.other.rules [hunk 1]
```diff
@@ -0,0 +1,8 @@
+rules_version = '2';
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/multiple-targets/storage.other.rules:1]: same comment as above

File: scripts/storage-emulator-integration/multiple-targets/tests.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const emulatorConfig = readEmulatorConfig(FIREBASE_EMULATOR_CONFIG);
+  const STORAGE_EMULATOR_HOST = getStorageEmulatorHost(emulatorConfig);
+
+  before(async function (this) {
+    this.timeout(TEST_SETUP_TIMEOUT);
+
+    test = new TriggerEndToEndTest(FIREBASE_PROJECT, __dirname, emulatorConfig);
+    await test.applyTargets(["storage", "main", storageBucket]);
+    await test.applyTargets(["storage", "other", otherStorageBucket]);
+    await test.startEmulators(["--only", "storage"]);
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/multiple-targets/tests.ts:26]: do we not want the auth emulator?

File: scripts/storage-emulator-integration/tests.ts [hunk 3]
```diff
@@ -6,24 +6,36 @@ import * as path from "path";
 import * as http from "http";
 import * as https from "https";
 import * as puppeteer from "puppeteer";
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/tests.ts:9]: Done.

File: scripts/storage-emulator-integration/multiple-targets/storage.main.rules [hunk 4]
```diff
@@ -0,0 +1,8 @@
+rules_version = '2';
```
[the remark this answers is not in the document]
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/multiple-targets/storage.main.rules:1]: Done.

File: scripts/storage-emulator-integration/multiple-targets/storage.other.rules (hunk 1 above)
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/multiple-targets/storage.other.rules:1]: Done.

File: scripts/storage-emulator-integration/multiple-targets/tests.ts (hunk 2 above)
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/multiple-targets/tests.ts:26]: Nope, no need to authenticate any requests. Either the rules allow all or allow none.

File: scripts/integration-helpers/framework.ts [hunk 5]
```diff
@@ -256,6 +256,18 @@ export class TriggerEndToEndTest {
     return this.cliProcess ? this.cliProcess.stop() : Promise.resolve();
   }
 
+  applyTargets(additionalArgs: string[]): Promise<void> {
```
[the remark this answers is not in the document]
Comment (tohhsinpei) [reply scripts/integration-helpers/framework.ts:259]: Done.

File: scripts/storage-emulator-integration/multiple-targets/tests.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  readEmulatorConfig,
+  TEST_SETUP_TIMEOUT,
+} from "../utils";
+
+const FIREBASE_PROJECT = process.env.FBTOOLS_TARGET_PROJECT || "fake-project-id";
+
+describe("Multiple Storage Deploy Targets", () => {
+  let test: TriggerEndToEndTest;
+  const storageBucket = `${FIREBASE_PROJECT}.appspot.com`;
+  const otherStorageBucket = `${FIREBASE_PROJECT}-2.appspot.com`;
```
[the remark this answers is not in the document]
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/multiple-targets/tests.ts:16]: Done.

**Review by tonyjhuang — APPROVED** [review]
Thanks for the change!

**Pull request merged**